test/unittest/dds/participant/ParticipantTests.cpp

```diff
@@ -376,6 +376,237 @@ TEST(ParticipantTests, ChangePSMDomainParticipantQos)
+    participant->get_qos(set_qos);
+    ASSERT_FALSE(set_qos == qos);
+
+    // Check changing wire_protocol().builtin.discovery_config.discoveryServer_client_syncperiod is NOT OK
+    participant->get_qos(qos);
+    qos.wire_protocol().builtin.discovery_config.discoveryServer_client_syncperiod = { 27, 27};
+    ASSERT_TRUE(participant->set_qos(qos) == ReturnCode_t::RETCODE_IMMUTABLE_POLICY);
+    participant->get_qos(set_qos);
+    ASSERT_FALSE(set_qos == qos);
+
+    // Check changing wire_protocol().builtin.discovery_config.leaseDuration is NOT OK
+    participant->get_qos(qos);
+    qos.wire_protocol().builtin.discovery_config.leaseDuration = { 27, 27};
+    ASSERT_TRUE(participant->set_qos(qos) == ReturnCode_t::RETCODE_IMMUTABLE_POLICY);
+    participant->get_qos(set_qos);
+    ASSERT_FALSE(set_qos == qos);
+
+    // Check changing wire_protocol().builtin.discovery_config.leaseDuration_announcementperiod is NOT OK
+    participant->get_qos(qos);
+    qos.wire_protocol().builtin.discovery_config.leaseDuration_announcementperiod = { 27, 27};
+    ASSERT_TRUE(participant->set_qos(qos) == ReturnCode_t::RETCODE_IMMUTABLE_POLICY);
+    participant->get_qos(set_qos);
+    ASSERT_FALSE(set_qos == qos);
+
+    // Check changing wire_protocol().builtin.discovery_config.initial_announcements is NOT OK
+    participant->get_qos(qos);
+    qos.wire_protocol().builtin.discovery_config.initial_announcements.count = 27;
+    qos.wire_protocol().builtin.discovery_config.initial_announcements.period = {27, 27};
+    ASSERT_TRUE(participant->set_qos(qos) == ReturnCode_t::RETCODE_IMMUTABLE_POLICY);
+    participant->get_qos(set_qos);
+    ASSERT_FALSE(set_qos == qos);
+
+    // Check changing wire_protocol().builtin.discovery_config.m_simpleEDP is NOT OK
+    participant->get_qos(qos);
+    qos.wire_protocol().builtin.discovery_config.m_simpleEDP.use_PublicationWriterANDSubscriptionReader ^= true;
+    qos.wire_protocol().builtin.discovery_config.m_simpleEDP.use_PublicationReaderANDSubscriptionWriter ^= true;
+#if HAVE_SECURITY
+    qos.wire_protocol().builtin.discovery_config.m_simpleEDP.enable_builtin_secure_publications_writer_and_subscriptions_reader ^= true;
+    qos.wire_protocol().builtin.discovery_config.m_simpleEDP.enable_builtin_secure_subscriptions_writer_and_publications_reader ^= true;
+#endif // if HAVE_SECURITY
+    ASSERT_TRUE(participant->set_qos(qos) == ReturnCode_t::RETCODE_IMMUTABLE_POLICY);
+    participant->get_qos(set_qos);
+    ASSERT_FALSE(set_qos == qos);
+
+    // Check changing wire_protocol().builtin.discovery_config.ignoreParticipantFlags is NOT OK
+    participant->get_qos(qos);
+    qos.wire_protocol().builtin.discovery_config.ignoreParticipantFlags = fastrtps::rtps::ParticipantFilteringFlags::FILTER_DIFFERENT_HOST;
+    ASSERT_TRUE(participant->set_qos(qos) == ReturnCode_t::RETCODE_IMMUTABLE_POLICY);
+    participant->get_qos(set_qos);
+    ASSERT_FALSE(set_qos == qos);
+
```

Comment:
Adding `DiscoverySettings::static_edp_xml_config_` check in the implementation will allow to check this change as well in this test.